gatling-core/src/main/scala/io/gatling/core/check/CheckBuilder.scala

```diff
@@ -175,6 +176,14 @@ private final case class DefaultValidatorCheckBuilder[T, P, X](extractor: Expres
   override def transformOptionWithSession[X2](transformation: (Option[X], Session) => Validation[Option[X2]]): ValidatorCheckBuilder[T, P, X2] =
     copy(extractor = session => extractor(session).map(transformOptionExtractor(transformation(_, session))))
 
+  override def withDefault(other: Expression[X]): ValidatorCheckBuilder[T, P, X] =
+    transformOptionWithSession((actual, session) =>
+      (actual match {
+        case Some(_) => actual
+        case _       => other(session).toOption
```

Comment:
Please reverse the cases to save an unapply:

case None => ???
case _ => ???

Reply:
Got it.